app/javascript/utilities/protectedFetch.js

```diff
@@ -0,0 +1,40 @@
+import {authenticatedFetch} from '@shopify/app-bridge-utils';
+import {AuthCode} from '@shopify/app-bridge/actions';
+
+function getAuthCode(appBridge) {
+  return new Promise((resolve, reject) => {
+    const unsubscribe = appBridge.subscribe(AuthCode.ActionType.RESPOND, (payload) => {
+      if (payload) {
+        resolve(payload);
+      } else {
+        reject("Failed to load");
+      }
+      unsubscribe();
+    });
+
+    appBridge.dispatch(AuthCode.request());
+  });
+}
```

Comment:
This is a great candidate for app-bridge. Do we want to expose this logic in this public demo app?